src/main/java/com/autotune/common/data/datasource/PrometheusDataSource.java

```diff
@@ -93,7 +99,12 @@ private String getValueForQuery(String response) throws IndexOutOfBoundsExceptio
 		} catch (UnsupportedEncodingException ignored) { }
 
 		String queryURL = dataSourceURL + getQueryEndpoint() + query;
-		String response = HttpUtils.getDataFromURL(new URL(queryURL), token);
+		String response = null;
+		try {
+			response = HttpUtils.getDataFromURL(new URL(queryURL), this.getToken());
+		} catch (IOException e) {
+			LOGGER.error("Unable to proceed due to invalid connection to URL: "+ queryURL);
```

Comment:
If there is a error here, we need to return null here instead of proceeding

Reply:
Sorry I meant that we should return in case of error and not continue. We dont need to return null, an empty list should be the right thing. As that is what we check in the calling function as well